arp_scripts/file_location.py

```diff
@@ -1,43 +1,69 @@
-#!/bin/python
+#!/usr/bin/env python
 
 import requests
 import json
 import argparse
 import logging
+import sys
 
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.DEBUG)
 
-parser = argparse.ArgumentParser()
-parser.add_argument('--run', '-r', type=int, action='store', help="Run number")
-parser.add_argument('--experiment', '-e', type=str, action='store', help="Experiment name")
 
-def get_file_location(exp, run):
+def get_file_location(exp, run, location):
     ws_url = f"https://pswww.slac.stanford.edu/ws/lgbk/lgbk/{exp}/ws/{run}/files_for_live_mode_at_location"
 
     # check sdf location
-    r = requests.get(ws_url, params={"location": "S3DF"})
+    r = requests.get(ws_url, params={"location": location})
     r.raise_for_status()
     on_sdf = r.json()['value']['all_present']
     
     if on_sdf:
```

Comment:
Now that location is an argument, should on_sdf still be called that and make the (fixed) path printout still sense?

Reply:
It's an argument for if we want to use that script elsewehere. For smd processing I think having it as S3DF always is fine.
The way it works is really just that http request. It return a list of the files and whether they are present or not, as well as a all_present flag that I read here. As I mentioned, I am preparing a case for when we want to ignoe stream 80 and above

Reply:
It's an argument for if we want to use that script elsewehere. For smd processing I think having it as S3DF always is fine.
The way it works is really just that http request. It return a list of the files and whether they are present or not, as well as a all_present flag that I read here. I also prepared a case for when we want to ignoe stream 80 and above. This needs a bit more work